.github/workflows/codeql.yml

```diff
@@ -122,7 +123,44 @@ jobs:
         if: ${{ matrix.goos == 'linux' }}
         run: make bin/vsockexec bin/init
 
+      # only upload results if the analysis fails
+      # otherwise, save the output and use `advanced-security/filter-sarif` to filter paths
       - name: CodeQL Analyze
         uses: github/codeql-action/analyze@v3
         with:
           category: "/language:${{matrix.language}}"
+          output: sarif-results
+          upload: failure-only
+
+      - name: Filter Go SARIF Results
+        uses: advanced-security/filter-sarif@v1
+        with:
+          patterns: |
+            +**/*.go
+            -**/*_test.go
+            -test/**/*.go
+            -vendor/**/*.go
```

Comment:
Don't we want the alerts about the dependencies we have?

Reply:
we can't really fix them tho
so at best, we can update them, but ideally the owner should be dealing with that